### ISSUE TITLE
Unable to fetch template body exception with cdklocal deploy

### ISSUE THREAD

**Issue by kenbarnettpt** [localstack/localstack#3706]
# Type of request: This is a ...

[X] bug report
[ ] feature request

# Detailed description
Deploying a CDK stack under cdklocal/localstack is throwing the following error:
`Exception: Unable to fetch template body (code 0) from URL https://s3.us-east-1.localhost/cdktoolkit-none-542d7434/assets/c69358cb0afe18dc5c97dc06107604b1a77f4e238de448f6286ab2bee582ceae.json`

Under WSL2 Debian, the following commands were executed:
`
docker-compose up -d
cdklocal boostrap aws://000000000000/us-east-1
cdklocal deploy orange
`

LocalStack PRO version: 0.12.7
LocalStack docker image id: 03a3685b80b8
LocalStack build date: 20210310
LocalStack build git hash: 95f91f6
aws-cdk version 1.93
aws-cdk-local version 1.65.3

The expanded stack trace:
`
*** FROM 'cdklocal bootstrap aws://000000000000/us-east-1'
2021-03-11T21:41:22:DEBUG:localstack.services.cloudformation.cloudformation_api: Executing change set "cdk-deploy-change-set" for stack "CDKToolkit" with 2 resources ...
2021-03-11T21:41:23:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: Parameter Ref
2021-03-11T21:41:23:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "StagingBucket" (1/2) type "AWS::S3::Bucket" in loop iteration 1
2021-03-11T21:41:23:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "S3::Bucket" id "StagingBucket"
2021-03-11T21:41:23:DEBUG:localstack.utils.cloudformation.template_deployer: Request for resource type "S3::Bucket" in region us-east-1: create_bucket {'Bucket': 'cdktoolkit-none-542d7434', 'ACL': 'private'}
2021-03-11T21:41:23:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "StagingBucketPolicy" (1/1) type "AWS::S3::BucketPolicy" in loop iteration 1
2021-03-11T21:41:23:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "S3::BucketPolicy" id "StagingBucketPolicy"
2021-03-11T21:41:23:DEBUG:localstack.utils.cloudformation.template_deployer: Request for resource type "S3::BucketPolicy" in region us-east-1: put_bucket_policy {'Bucket': 'cdktoolkit-none-542d7434', 'Policy': '{"Id": "AccessControl", "Version": "2012-10-17", "Statement": [{"Sid": "AllowSSLRequestsOnly", "Action": "s3:*", "Effect": "Deny", "Resource": ["arn:aws:s3:::cdktoolkit-none-542d7434", "arn:aws:s3:::cdktoolkit-none-542d7434/*"], "Condition": {"Bool": {"aws:SecureTransport": "false"}}, "Principal": "*"}]}'}
2021-03-11T21:41:23:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: S3::BucketPolicy PhysicalResourceId
2021-03-11T21:41:24:INFO:localstack.utils.cloudformation.template_deployer: Unable to determine PhysicalResourceId for "S3::BucketPolicy" resource, ID "StagingBucketPolicy"
2021-03-11T21:41:24:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: S3::Bucket RegionalDomainName
*** FROM 'cdklocal deploy orange'
2021-03-11T21:42:05:DEBUG:localstack.services.cloudformation.cloudformation_api: Executing change set "cdk-deploy-change-set" for stack "orange" with 35 resources ...
2021-03-11T21:42:05:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "xyzorangegluecrawlerrole91E4E6D6" (1/35) type "AWS::IAM::Role" in loop iteration 1
2021-03-11T21:42:05:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "IAM::Role" id "xyzorangegluecrawlerrole91E4E6D6"
2021-03-11T21:42:05:DEBUG:localstack.utils.cloudformation.template_deployer: Request for resource type "IAM::Role" in region us-east-1: create_role {'AssumeRolePolicyDocument': '{"Statement": [{"Action": "sts:AssumeRole", "Effect": "Allow", "Principal": {"Service": "glue.amazonaws.com"}}], "Version": "2012-10-17"}', 'RoleName': 'xyz-orange-glue-crawler-role'}
2021-03-11T21:42:06:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "xyzorangegluecrawlerroleDefaultPolicy9ED9F5DD" (1/34) type "AWS::IAM::Policy" in loop iteration 1
2021-03-11T21:42:06:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "IAM::Policy" id "xyzorangegluecrawlerroleDefaultPolicy9ED9F5DD"
2021-03-11T21:42:06:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "xyzorangeredshiftspectrumroleAF1ADF87" (1/33) type "AWS::IAM::Role" in loop iteration 1
2021-03-11T21:42:06:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "IAM::Role" id "xyzorangeredshiftspectrumroleAF1ADF87"
2021-03-11T21:42:06:DEBUG:localstack.utils.cloudformation.template_deployer: Request for resource type "IAM::Role" in region us-east-1: create_role {'AssumeRolePolicyDocument': '{"Statement": [{"Action": "sts:AssumeRole", "Effect": "Allow", "Principal": {"Service": "redshift.amazonaws.com"}}], "Version": "2012-10-17"}', 'RoleName': 'xyz-orange-redshift-spectrum-role'}
2021-03-11T21:42:06:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "xyzorangeredshiftspectrumroleDefaultPolicy3F037135" (1/32) type "AWS::IAM::Policy" in loop iteration 1
2021-03-11T21:42:06:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "IAM::Policy" id "xyzorangeredshiftspectrumroleDefaultPolicy3F037135"
2021-03-11T21:42:07:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "xyzorangeredshiftinitializerroleDD2C659D" (1/31) type "AWS::IAM::Role" in loop iteration 1
2021-03-11T21:42:07:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "IAM::Role" id "xyzorangeredshiftinitializerroleDD2C659D"
2021-03-11T21:42:07:DEBUG:localstack.utils.cloudformation.template_deployer: Request for resource type "IAM::Role" in region us-east-1: create_role {'AssumeRolePolicyDocument': '{"Statement": [{"Action": "sts:AssumeRole", "Effect": "Allow", "Principal": {"Service": "redshift.amazonaws.com"}}], "Version": "2012-10-17"}', 'RoleName': 'xyz-orange-redshift-initializer-role'}
2021-03-11T21:42:07:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "xyzorangeredshiftinitializerroleDefaultPolicyBFFE5036" (1/30) type "AWS::IAM::Policy" in loop iteration 1
2021-03-11T21:42:07:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "IAM::Policy" id "xyzorangeredshiftinitializerroleDefaultPolicyBFFE5036"
2021-03-11T21:42:07:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "xyzoranges3replicationroleA59F77AF" (1/29) type "AWS::IAM::Role" in loop iteration 1
2021-03-11T21:42:07:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "IAM::Role" id "xyzoranges3replicationroleA59F77AF"
2021-03-11T21:42:07:DEBUG:localstack.utils.cloudformation.template_deployer: Request for resource type "IAM::Role" in region us-east-1: create_role {'AssumeRolePolicyDocument': '{"Statement": [{"Action": "sts:AssumeRole", "Effect": "Allow", "Principal": {"Service": "glue.amazonaws.com"}}], "Version": "2012-10-17"}', 'RoleName': 'xyz-orange-s3-replication-role'}
2021-03-11T21:42:08:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "xyzoranges3replicationroleDefaultPolicy8CDB2912" (1/28) type "AWS::IAM::Policy" in loop iteration 1
2021-03-11T21:42:08:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "IAM::Policy" id "xyzoranges3replicationroleDefaultPolicy8CDB2912"
2021-03-11T21:42:08:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "scriptsorange65D490A4" (1/27) type "AWS::S3::Bucket" in loop iteration 1
2021-03-11T21:42:08:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "S3::Bucket" id "scriptsorange65D490A4"
2021-03-11T21:42:08:DEBUG:localstack.utils.cloudformation.template_deployer: Request for resource type "S3::Bucket" in region us-east-1: create_bucket {'Bucket': 'aws-glue-ae-orange-scripts-000000000000-us-east-1', 'ACL': 'public-read'}
2021-03-11T21:42:09:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "scriptsorangePolicyC0BC71F3" (1/26) type "AWS::S3::BucketPolicy" in loop iteration 1
2021-03-11T21:42:09:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "S3::BucketPolicy" id "scriptsorangePolicyC0BC71F3"
2021-03-11T21:42:09:DEBUG:localstack.utils.cloudformation.template_deployer: Request for resource type "S3::BucketPolicy" in region us-east-1: put_bucket_policy {'Bucket': 'aws-glue-ae-orange-scripts-000000000000-us-east-1', 'Policy': '{"Statement": [{"Action": "s3:*", "Effect": "Allow", "Principal": {"AWS": "arn:aws:iam::000000000000:root"}, "Resource": "arn:aws:s3:::aws-glue-ae-orange-scripts-000000000000-us-east-1"}, {"Action": "s3:*", "Effect": "Allow", "Principal": {"AWS": ["arn:aws:iam::000000000000:role/xyz-orange-glue-crawler-role", "arn:aws:iam::000000000000:role/xyz-orange-redshift-spectrum-role", "arn:aws:iam::000000000000:role/xyz-orange-redshift-initializer-role", "arn:aws:iam::000000000000:role/xyz-orange-s3-replication-role"]}, "Resource": "arn:aws:s3:::aws-glue-ae-orange-scripts-000000000000-us-east-1"}], "Version": "2012-10-17"}'}
2021-03-11T21:42:09:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: S3::BucketPolicy PhysicalResourceId
2021-03-11T21:42:09:INFO:localstack.utils.cloudformation.template_deployer: Unable to determine PhysicalResourceId for "S3::BucketPolicy" resource, ID "scriptsorangePolicyC0BC71F3"
2021-03-11T21:42:09:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "temporange173DA744" (1/25) type "AWS::S3::Bucket" in loop iteration 1
2021-03-11T21:42:09:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "S3::Bucket" id "temporange173DA744"
2021-03-11T21:42:09:DEBUG:localstack.utils.cloudformation.template_deployer: Request for resource type "S3::Bucket" in region us-east-1: create_bucket {'Bucket': 'aws-glue-ae-orange-temp-000000000000-us-east-1', 'ACL': 'public-read'}
2021-03-11T21:42:10:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "temporangePolicy96650AC0" (1/24) type "AWS::S3::BucketPolicy" in loop iteration 1
2021-03-11T21:42:10:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "S3::BucketPolicy" id "temporangePolicy96650AC0"
2021-03-11T21:42:10:DEBUG:localstack.utils.cloudformation.template_deployer: Request for resource type "S3::BucketPolicy" in region us-east-1: put_bucket_policy {'Bucket': 'aws-glue-ae-orange-temp-000000000000-us-east-1', 'Policy': '{"Statement": [{"Action": "s3:*", "Effect": "Allow", "Principal": {"AWS": "arn:aws:iam::000000000000:root"}, "Resource": "arn:aws:s3:::aws-glue-ae-orange-temp-000000000000-us-east-1"}, {"Action": "s3:*", "Effect": "Allow", "Principal": {"AWS": ["arn:aws:iam::000000000000:role/xyz-orange-glue-crawler-role", "arn:aws:iam::000000000000:role/xyz-orange-redshift-spectrum-role", "arn:aws:iam::000000000000:role/xyz-orange-redshift-initializer-role", "arn:aws:iam::000000000000:role/xyz-orange-s3-replication-role"]}, "Resource": "arn:aws:s3:::aws-glue-ae-orange-temp-000000000000-us-east-1"}], "Version": "2012-10-17"}'}
2021-03-11T21:42:10:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: S3::BucketPolicy PhysicalResourceId
2021-03-11T21:42:10:INFO:localstack.utils.cloudformation.template_deployer: Unable to determine PhysicalResourceId for "S3::BucketPolicy" resource, ID "temporangePolicy96650AC0"
2021-03-11T21:42:10:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: Parameter Ref
2021-03-11T21:42:10:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: Parameter Ref
2021-03-11T21:42:10:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: Parameter Ref
2021-03-11T21:42:10:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "orangeinitNestedStackorangeinitNestedStackResource43AB251E" (1/23) type "AWS::CloudFormation::Stack" in loop iteration 1
2021-03-11T21:42:10:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "CloudFormation::Stack" id "orangeinitNestedStackorangeinitNestedStackResource43AB251E"
2021-03-11T21:42:10:DEBUG:localstack.utils.cloudformation.template_deployer: Request for resource type "CloudFormation::Stack" in region us-east-1: create_stack {'StackName': 'orange-None-d435f146', 'TemplateURL': 'https://s3.us-east-1.localhost/cdktoolkit-none-542d7434/assets/c69358cb0afe18dc5c97dc06107604b1a77f4e238de448f6286ab2bee582ceae.json', 'Parameters': [{'ParameterKey': 'referencetoorangescriptsorangeC008DBC1Ref', 'ParameterValue': 'aws-glue-ae-orange-scripts-000000000000-us-east-1'}]}
2021-03-11T21:42:11:ERROR:cloudformation_api: Exception on / [POST]
Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 549, in handle_request
    result = func(req_params)
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 304, in create_stack
    template_deployer.prepare_template_body(req_params)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1183, in prepare_template_body
    return template_preparer.prepare_template_body(req_data)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_preparer.py", line 63, in prepare_template_body
    modified_template_body = transform_template(req_data)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_preparer.py", line 26, in transform_template
    template_body = get_template_body(req_data)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_preparer.py", line 127, in get_template_body
    raise Exception('Unable to fetch template body (code %s) from URL %s' % (status_code, url))
Exception: Unable to fetch template body (code 0) from URL https://s3.us-east-1.localhost/cdktoolkit-none-542d7434/assets/c69358cb0afe18dc5c97dc06107604b1a77f4e238de448f6286ab2bee582ceae.json
2021-03-11T21:42:11:WARNING:localstack.utils.cloudformation.template_deployer: Error calling <bound method ClientCreator._create_api_method.<locals>._api_call of <botocore.client.CloudFormation object at 0x7fd22adc9400>> with params: {'StackName': 'orange-None-d435f146', 'TemplateURL': 'https://s3.us-east-1.localhost/cdktoolkit-none-542d7434/assets/c69358cb0afe18dc5c97dc06107604b1a77f4e238de448f6286ab2bee582ceae.json', 'Parameters': [{'ParameterKey': 'referencetoorangescriptsorangeC008DBC1Ref', 'ParameterValue': 'aws-glue-ae-orange-scripts-000000000000-us-east-1'}]} for resource: {'Type': 'AWS::CloudFormation::Stack', 'UpdateReplacePolicy': 'Delete', 'DeletionPolicy': 'Delete', 'Metadata': {'aws:cdk:path': 'orange/orange_init.NestedStack/orange_init.NestedStackResource'}, 'LogicalResourceId': 'orangeinitNestedStackorangeinitNestedStackResource43AB251E', 'Properties': {'TemplateURL': 'https://s3.us-east-1.localhost/cdktoolkit-none-542d7434/assets/c69358cb0afe18dc5c97dc06107604b1a77f4e238de448f6286ab2bee582ceae.json', 'Parameters': {'referencetoorangescriptsorangeC008DBC1Ref': 'aws-glue-ae-orange-scripts-000000000000-us-east-1'}, 'StackName': 'orange-None-d435f146'}, '_state_': {}}
`
## Expected behavior
The CloudFormation template is generated by cdk, resources created/mutated, and the stack created.

This behaves as expected under cdk w/o localstack.

## Actual behavior
The previously described exception is thrown.

# Steps to reproduce

## Command used to start LocalStack
docker-compose up -d

The docker-compose file is:
`
version: '2.1'

services:
  localstack:
    container_name: "${LOCALSTACK_DOCKER_NAME-localstack_main}"
    image: localstack/localstack
    network_mode: bridge
    ports:
      - "4566:4566"
      - "4571:4571"
      - "443:443"
      - "${LOCALSTACK_PORT_WEB_UI-8080}:${PORT_WEB_UI-8080}"
    environment:
      - TEST_AWS_ACCOUNT_ID=000000000000
      - AWS_REGION=US-EAST-1
      - AWS_ACCOUNT_ID=000000000000
      - LOCALSTACK_API_KEY=<secret>
      - SERVICES=${SERVICES- }
      - DEBUG=${LOCALSTACK_DEBUG- }
      - DATA_DIR=${LOCALSTACK_DATA_DIR- }
      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR- }
      - KINESIS_ERROR_PROBABILITY=${KINESIS_ERROR_PROBABILITY- }
      - DOCKER_HOST=unix:///var/run/docker.sock
      - HOST_TMP_FOLDER=${LOCALSTACK_TMP_DIR}
    volumes:
      - "./data:/tmp/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"
      - "./bin:/docker-entrypoint-initaws.d"
`

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

...

**Comment by magno32** [comment]
I also have this issue.  To me it looks like an issue with nested stacks.  I've included a simple reproducible example.

1. Start localstack: `docker run --rm -p 4566:4566 -p 4571:4571 -p 8080:8080 -e LOCALSTACK_API_KEY=******  -e DEBUG=1 localstack/localstack-full`
2. `cdklocal bootstrap`
3. `cdklocal deploy ErrorExampleStack` *SUCCESS*
4. `cdklocal deploy ErrorExampleNestedStack` _**FAILS**_

[error_example.zip](https://github.com/localstack/localstack/files/6257904/error_example.zip)

Output after startup:

```
2021-04-05T11:36:50:DEBUG:localstack.services.cloudformation.cloudformation_api: Executing change set "cdk-deploy-change-set" for stack "CDKToolkit" with 2 resources ...
2021-04-05T11:36:51:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: Parameter Ref
2021-04-05T11:36:51:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "StagingBucket" (1/2) type "AWS::S3::Bucket" in loop iteration 1
2021-04-05T11:36:51:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "S3::Bucket" id "StagingBucket"
2021-04-05T11:36:51:DEBUG:localstack.utils.cloudformation.template_deployer: Request for resource type "S3::Bucket" in region us-east-1: create_bucket {'Bucket': 'cdktoolkit-none-f83add60', 'ACL': 'private'}
2021-04-05T11:36:51:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: S3::Bucket RegionalDomainName
2021-04-05T11:36:51:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "StagingBucketPolicy" (1/1) type "AWS::S3::BucketPolicy" in loop iteration 1
2021-04-05T11:36:51:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "S3::BucketPolicy" id "StagingBucketPolicy"
2021-04-05T11:36:51:DEBUG:localstack.utils.cloudformation.template_deployer: Request for resource type "S3::BucketPolicy" in region us-east-1: put_bucket_policy {'Bucket': 'cdktoolkit-none-f83add60', 'Policy': '{"Id": "AccessControl", "Version": "2012-10-17", "Statement": [{"Sid": "AllowSSLRequestsOnly", "Action": "s3:*", "Effect": "Deny", "Resource": ["arn:aws:s3:::cdktoolkit-none-f83add60", "arn:aws:s3:::cdktoolkit-none-f83add60/*"], "Condition": {"Bool": {"aws:SecureTransport": "false"}}, "Principal": "*"}]}'}
2021-04-05T11:36:52:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: S3::BucketPolicy PhysicalResourceId
2021-04-05T11:36:52:INFO:localstack.utils.cloudformation.template_deployer: Unable to determine PhysicalResourceId for "S3::BucketPolicy" resource, ID "StagingBucketPolicy"
2021-04-05T11:37:02:DEBUG:localstack.services.cloudformation.cloudformation_api: Executing change set "cdk-deploy-change-set" for stack "ErrorExampleStack" with 1 resources ...
2021-04-05T11:37:02:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: Parameter Ref
2021-04-05T11:37:02:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "CDKMetadata" (1/1) type "AWS::CDK::Metadata" in loop iteration 1
2021-04-05T11:37:02:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: Parameter Ref
2021-04-05T11:37:02:INFO:localstack.utils.cloudformation.template_deployer: Action "create" for resource type CDK::Metadata not yet implemented - using fallback mechanism
2021-04-05T11:37:02:DEBUG:localstack.utils.cloudformation.template_deployer: Unable to extract name for resource type "AWS::CDK::Metadata"
2021-04-05T11:37:02:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: CDK::Metadata PhysicalResourceId
2021-04-05T11:37:12:DEBUG:localstack.services.cloudformation.cloudformation_api: Executing change set "cdk-deploy-change-set" for stack "ErrorExampleNestedStack" with 2 resources ...
2021-04-05T11:37:12:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: Parameter Ref
2021-04-05T11:37:12:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: Parameter Ref
2021-04-05T11:37:12:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: Parameter Ref
2021-04-05T11:37:12:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "NestedStackNestedStackNestedStackNestedStackResourceB70834FD" (1/2) type "AWS::CloudFormation::Stack" in loop iteration 1
2021-04-05T11:37:12:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "create" for resource type "CloudFormation::Stack" id "NestedStackNestedStackNestedStackNestedStackResourceB70834FD"
2021-04-05T11:37:12:DEBUG:localstack.utils.cloudformation.template_deployer: Request for resource type "CloudFormation::Stack" in region us-east-1: create_stack {'StackName': 'ErrorExampleNestedStack-None-54c6dfe7', 'TemplateURL': 'https://s3.us-east-1.localhost/cdktoolkit-none-f83add60/assets/15fd998f92dbf2500f229f25319dfc502cea10df9b32a315b15780ed09ae3efb.json', 'Parameters': []}
2021-04-05T11:37:12:ERROR:cloudformation_api: Exception on / [POST]
Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 579, in handle_request
    result = func(req_params)
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_api.py", line 314, in create_stack
    template_deployer.prepare_template_body(req_params)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1093, in prepare_template_body
    return template_preparer.prepare_template_body(req_data)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_preparer.py", line 63, in prepare_template_body
    modified_template_body = transform_template(req_data)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_preparer.py", line 26, in transform_template
    template_body = get_template_body(req_data)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_preparer.py", line 127, in get_template_body
    raise Exception('Unable to fetch template body (code %s) from URL %s' % (status_code, url))
Exception: Unable to fetch template body (code 0) from URL https://s3.us-east-1.localhost/cdktoolkit-none-f83add60/assets/15fd998f92dbf2500f229f25319dfc502cea10df9b32a315b15780ed09ae3efb.json
2021-04-05T11:37:12:WARNING:localstack.utils.cloudformation.template_deployer: Error calling <bound method ClientCreator._create_api_method.<locals>._api_call of <botocore.client.CloudFormation object at 0x7f1c4d0f6070>> with params: {'StackName': 'ErrorExampleNestedStack-None-54c6dfe7', 'TemplateURL': 'https://s3.us-east-1.localhost/cdktoolkit-none-f83add60/assets/15fd998f92dbf2500f229f25319dfc502cea10df9b32a315b15780ed09ae3efb.json', 'Parameters': []} for resource: {'Type': 'AWS::CloudFormation::Stack', 'UpdateReplacePolicy': 'Delete', 'DeletionPolicy': 'Delete', 'Metadata': {'aws:cdk:path': 'ErrorExampleNestedStack/NestedStack.NestedStack/NestedStack.NestedStackResource'}, 'LogicalResourceId': 'NestedStackNestedStackNestedStackNestedStackResourceB70834FD', 'Properties': {'TemplateURL': 'https://s3.us-east-1.localhost/cdktoolkit-none-f83add60/assets/15fd998f92dbf2500f229f25319dfc502cea10df9b32a315b15780ed09ae3efb.json', 'StackName': 'ErrorExampleNestedStack-None-54c6dfe7'}, '_state_': {}}
2021-04-05T11:37:12:DEBUG:localstack.utils.cloudformation.template_deployer: Error applying changes for CloudFormation stack "ErrorExampleNestedStack": Unable to parse response (syntax error: line 1, column 54), invalid XML received. Further retries may succeed:
b'<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">\n<title>500 Internal Server Error</title>\n<h1>Internal Server Error</h1>\n<p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p>\n' Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/parsers.py", line 451, in _parse_xml_string_to_dom
    parser.feed(xml_string)
  File "<string>", line None
xml.etree.ElementTree.ParseError: syntax error: line 1, column 54

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1952, in _run
    self.do_apply_changes_in_loop(changes, stack, stack_name)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1999, in do_apply_changes_in_loop
    self.apply_change(change, stack, new_resources, stack_name=stack_name)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 2052, in apply_change
    result = deploy_resource(resource_id, new_resources, stack_name)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1097, in deploy_resource
    return execute_resource_action(resource_id, resources, stack_name, ACTION_CREATE)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1201, in execute_resource_action
    result = configure_resource_via_sdk(resource_id, resources, resource_type, func, stack_name, action_name)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1312, in configure_resource_via_sdk
    raise e
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1307, in configure_resource_via_sdk
    result = function(**params)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 357, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 662, in _make_api_call
    http, parsed_response = self._make_request(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 682, in _make_request
    return self._endpoint.make_request(operation_model, request_dict)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 102, in make_request
    return self._send_request(request_dict, operation_model)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 134, in _send_request
    success_response, exception = self._get_response(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 166, in _get_response
    success_response, exception = self._do_get_response(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/endpoint.py", line 217, in _do_get_response
    parsed_response = parser.parse(
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/parsers.py", line 243, in parse
    parsed = self._do_error_parse(response, shape)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/parsers.py", line 505, in _do_error_parse
    root = self._parse_xml_string_to_dom(xml_contents)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/parsers.py", line 454, in _parse_xml_string_to_dom
    raise ResponseParserError(
botocore.parsers.ResponseParserError: Unable to parse response (syntax error: line 1, column 54), invalid XML received. Further retries may succeed:
b'<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">\n<title>500 Internal Server Error</title>\n<h1>Internal Server Error</h1>\n<p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p>\n'
```

**Comment by whummer** [comment]
Thanks for reporting @kenbarnettpt @magno32 . This should (hopefully) be fixed in #3830 - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks!

**Comment by magno32** [comment]
@whummer This appears to have fixed that issue.

However, I'm getting more errors as well (that appear unrelated to me).  I'll do some digging an pin a new ticket if needed.

Thanks for pushing this.
